fakesnow/transforms.py

```diff
@@ -22,6 +22,32 @@ def array_size(expression: exp.Expression) -> exp.Expression:
     return expression
 
 
+def array_agg_within_group(expression: exp.Expression) -> exp.Expression:
+    """Convert ARRAY_AGG(<expr>) WITHIN GROUP (<order-by-clause>) to ARRAY_AGG( <expr> <order-by-clause> )
+    Snowflake uses ARRAY_AGG(<expr>) WITHIN GROUP (ORDER BY <order-by-clause>)
+    to order the array, but DuckDB uses ARRAY_AGG( <expr> <order-by-clause> ).
+    See;
+        - https://docs.snowflake.com/en/sql-reference/functions/array_agg
+        - https://duckdb.org/docs/sql/aggregates.html#order-by-clause-in-aggregate-functions
+    Note; Snowflake has following restriction;
+            If you specify DISTINCT and WITHIN GROUP, both must refer to the same column.
+          Transformation does not handle this restriction.
+    """
+    if (
+        isinstance(expression, exp.WithinGroup)
+        and (agg := expression.find(exp.ArrayAgg))
+        and (order := expression.expression)
+    ):
+        return exp.ArrayAgg(
+            this=exp.Order(
+                this=agg.this,
+                expressions=order.expressions,
+            )
+        )
```

Comment:
to produce a json array
```suggestion
        return exp.Anonymous(this="TO_JSON", expressions=[exp.ArrayAgg(
            this=exp.Order(
                this=agg.this,
                expressions=order.expressions,
            )
        )])
 ```

Reply:
missed this. besides this case, all `ArrayAgg`'s might need to be wrapped in `TO_JSON`. 

how about adding another transform that runs after this one that wraps `ArrayAgg`s?

Reply:
☝️ what i meant was something like this;
https://gist.github.com/seruman/79b050a0ff96c90d616350a2ca62dfaf

Reply:
Ah yep!

Reply:
thanks you, updated @https://github.com/tekumara/fakesnow/pull/66/commits/3c310cc14ac317698e4e19e404ea71c13c387e98